src/NextGen/DonationForm/DataTransferObjects/DonateFormRouteData.php

```diff
@@ -83,7 +85,17 @@ public function validated(): DonateControllerData
         }
 
         foreach ($validator->validated() as $fieldId => $value) {
-            $validData->{$fieldId} = $value;
+            switch ($fieldId) {
+                case 'donationType':
+                    $validData->{$fieldId} = new DonationType($value);
+                    break;
+                case 'subscriptionPeriod':
+                    $validData->{$fieldId} = $value ? new SubscriptionPeriod($value) : null;
+                    break;
```

Comment:
Why not make sanitizing `DonationType` and `SubscriptionType` validation rules?

Reply:
There are rules for both, is it possible to return the value objects in the rule instead of doing this?

Reply:
Ah I just figured it out abc891da

That's a cool feature, I didn't know that existed (or did I? 🤷 )